app/models/cim_base_storage_extent.rb

```diff
@@ -10,10 +10,7 @@ def self._virtual_reflections
   end
 
   def self.find(*args)
-    options = args.extract_options!
-    options[:conditions] = CimStorageExtent.merge_conditions(options[:conditions], :id => base_storage_extent_ids)
-    args << options
-    CimStorageExtent.find(*args)
+    CimStorageExtent.where(:id => base_storage_ext_ids).find(*args)
```

Comment:
ok, so it was pulling out the `options` hash out of `args` array, modifying the `:condition`, and then putting the `options` hash back into `args` array.

Per this PR, this will no longer modify `options`, but use a `where` clause to augment the conditions.
So if we are no longer modifying `options[:condition]` there is no reason to pull it out, or even disturb `args` at all.

But the `args` are still getting passed into find, so `:distinct`, `:conditions`, and all else will still flow through.